From: Julia Lawall <julia.lawall@inria.fr>
Date: Fri, 24 Jul 2020 12:05:31 +0200
Subject: [PATCH] printk: fix ifnullfree.cocci warnings
Origin: https://www.kernel.org/pub/linux/kernel/projects/rt/5.9/older/patches-5.9-rt16.tar.xz

Make the code a little simpler by dropping
some unneeded tests.

Generated by: scripts/coccinelle/free/ifnullfree.cocci

Fixes: c406fbce2054 ("printk: implement syslog")
CC: John Ogness <john.ogness@linutronix.de>
Signed-off-by: kernel test robot <lkp@intel.com>
Signed-off-by: Sebastian Andrzej Siewior <bigeasy@linutronix.de>
Acked-by: John Ogness <john.ogness@linutronix.de>
---
 kernel/printk/printk.c |   12 ++++--------
 1 file changed, 4 insertions(+), 8 deletions(-)

--- a/kernel/printk/printk.c
+++ b/kernel/printk/printk.c
@@ -1466,10 +1466,8 @@ static int syslog_print_all(char __user
 	if (clear && !seq)
 		syslog_clear();
 
-	if (text)
-		kfree(text);
-	if (msgbuf)
-		kfree(msgbuf);
+	kfree(text);
+	kfree(msgbuf);
 	return len;
 }
 
@@ -1622,10 +1620,8 @@ int do_syslog(int type, char __user *buf
 		break;
 	}
 out:
-	if (msgbuf)
-		kfree(msgbuf);
-	if (text)
-		kfree(text);
+	kfree(msgbuf);
+	kfree(text);
 	return error;
 }
 
